lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -2145,6 +2168,102 @@ def construct_error_details_map_grpc(gax_error)
       {}
     end
 
+    # Construct monitored resource locally for k8s resources.
+    def contruct_k8s_resource_locally(local_resource_id)
+      matched_regexp_group = nil
+      # Regex match the local_resource_id to make sure we only do
+      # this for 'k8s_container' and 'k8s_node'.
+      [@compiled_k8s_node_local_resource_id_regexp,
```

Comment:
I may have made it sound more dire than it is. The magic is only in defining local variables for regexp captures. It's "the Ruby way™".
That said, this also proposes a refactoring that goes beyond the local binding for captures.

Reply:
Yeah, it's not too "black magic". I do like the regex syntax.

But as Igor said, we have to be very literal (I did not realize what that meant until I gave it a try LOL). We have to have everything in pretty much one method (otherwise the conciseness is not sustained).

Will hold on for this comment before I finish unit tests. Will probably implement whichever way that's easier to unit test.

Reply:
I actually like the fact that all of the relevant code and constants are encapsulated within the same method. The more globals we add, the more locations there would be to clean up later (and TODOs in the code). So I would advocate for the one method approach (with the added benefit of the locals). This would also be the only method to test.

Reply:
Refactored. PTAL. I changed the order of minor things because the error messages are different (unless we want a hash approach. I kind of feel like that's an overkill).